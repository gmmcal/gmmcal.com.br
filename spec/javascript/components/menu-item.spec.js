import { mount } from '@vue/test-utils'
import MenuItem from '@/components/menu-item'
import create from '../settings/utils'

describe('MenuItem', () => {
  let props, wrapper

  beforeEach(() => {
    props = {
      propsData: {
        item: create('menu_item')
      }
    }

    wrapper = mount(MenuItem, props)
  })

  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('has data properties', () => {
    const expected = ['I18n', 'click']
    const received = Object.keys(wrapper.vm.$data)
    expect(received).toEqual(expected)
  })

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
