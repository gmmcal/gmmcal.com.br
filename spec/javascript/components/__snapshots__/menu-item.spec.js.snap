// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem renders correctly 1`] = `<a data-turbolinks="false" href="#home" alt="Home" class="home">[missing "en.template.menu.Home" translation]</a>`;
